src/repr/src/adt/regex.rs

```diff
@@ -21,6 +21,17 @@ use serde::de::Error as DeError;
 use serde::ser::SerializeStruct;
 use serde::{Deserialize, Deserializer, Serialize, Serializer, de};
 
+/// The maximum size of a regex after compilation.
+/// This is the same as the `Regex` crate's default at the time of writing.
+const MAX_REGEX_SIZE_AFTER_COMPILATION: usize = 10 * 1024 * 1024;
+
+/// We also need a separate limit for the size of regexes before compilation. Even though the
+/// `Regex` crate promises that using its `size_limit` option (which we set to the other limit,
+/// `MAX_REGEX_SIZE_AFTER_COMPILATION`) would prevent excessive resource usage, this doesn't seem to
+/// be the case. Since we compile regexes in envd, we need strict limits to prevent envd OOMs.
+/// See <https://github.com/MaterializeInc/database-issues/issues/9907> for an example.
+const MAX_REGEX_SIZE_BEFORE_COMPILATION: usize = 1 * 1024 * 1024;
```

Comment:
Should we document this constant in our docs?

Is it a potential regression that we don't accept a query that works fine at the moment?

Reply:
We can simply check this for existing views using mzexplore---should be quick to export data and run some `jq` queries.